### PR TITLE
support for a logo field in a package.json

### PR DESCRIPTION
**Pull request by bcoe**
we'll now look for a logo field in package.json and use it if it's a string:

screen shot 2016-02-03 at 3 05 36 pm

CC: @jefflembeck

### CODE REVIEW

File: test/fixtures/data.json [hunk 1]
```diff
@@ -41342,7 +41342,8 @@
         ]
       },
       "10.2.6": {
-        "name": "browserify",
+       	"logo": "http://example.com/logo.png",
+	"name": "browserify",
```
Comment (nexdrew) [inline test/fixtures/data.json:41346]: [nit] Some tabs on these 2 lines instead of spaces :)

**Comment by nexdrew** [comment]
Assuming support for URLs only (instead of package-local raster files), this LGTM. :+1:

**Comment by seldo** [comment]
I have performance and security concerns about unvalidated arbitrary URLs being displayed on the home page. In particular, SVG files are XML and as such can contain executable code including JavaScript, and so are critical XSS vulns [(source)](http://security.stackexchange.com/questions/11384/exploits-or-other-security-risks-with-svg-upload).

This is still a neat feature though, so we can let it go in a safer, npm On-Site context if:
- it is behind an npmO feature flag
- it limits itself to {png,gif,jpeg} files and renders them as images with fixed sizes
- it can be switched off in npmO configuration quickly should a vuln present itself

Longer-term it should validate images and mirror them locally before display. Depending how much we can lock this down it could be taken to the public site, pending a bikeshed about how to support multiple image sizes and resolutions.

**Comment by marcellodesales** [comment]
@bcoe This is a great start for us! Definitely what we are looking for! 

@seldo Good points... Maybe we can restrict the logos from URLs based on the npmo host in the case of `svg` icons. For instance, logos from `github.mycompany.com/o/r/mylogo.svg` should be allowed, but not `github.com/r/o/logo.svg`.

**Comment by bcoe** [comment]
@marcellodesales this seems like a good `v2` feature, one idea @seldo floated which I think is smart -- is eventually we have our own image-hosting service that serves blessed logos. I could imagine us shipping this behind the firewall.

**Comment by seldo** [comment]
We already wrote one, back when Z was around: https://github.com/zeke/npm-cdn

**Comment by kevinSuttle** [comment]
So, I'm going to do a couple of things here. 
1. Loop in @mjackson, who has built and maintained https://npmcdn.com.
2. Point out the lengthy prior discussions in support of this:
   https://github.com/schemaorg/schemaorg/issues/939
   including some from npm which are already out in the wild 
   https://github.com/npm/newww/issues/291
   https://github.com/npm/newww/issues/286

**Comment by seldo** [comment]
Creating a CDN that can unpack and serve package data is not primarily a technical challenge (as you note, there are a couple extant implementations) but, in order of priority:
1. security
2. performance
3. aesthetics

That schema suggestion certainly seems to have done a lot of the bikeshedding in advance. The perf angle is handled if we pre-fetch and cache the content or serve it directly out of package files. So the primary concern is security, which unfortunately rules out SVGs, because SVGs solve a lot of aesthetic problems related to sizes & resolutions.

**Comment by kevinSuttle** [comment]
@seldo That's super interesting. I've never heard of SVG being used for embedded script execution, but that is certainly a reality. I have to think there's a sensible away around that. Unless we're serving photos, it feels short-sighted to serve rastered graphics. 

@bcoe had a great idea too. 

> There exist CDNs that resize images, we could encourage that to be a standard approach, rather than complicating the logo stanza. 
> 
> org.cdn.com/foo.png?width=…

This same service could easily parse and reject any embedded scripts, _but_, that would also negate some of the more advanced responsive SVG techniques. 

https://www.smashingmagazine.com/2014/03/rethinking-responsive-svg/

Or even upon publish npm itself could parse the SVG DOM, and reject anything it deems dangerous.

**Comment by kevinSuttle** [comment]
For reference: https://code.google.com/p/chromium/issues/detail?id=470601

**Comment by seldo** [comment]
It is possible ([source](http://stackoverflow.com/questions/11632748/svg-to-png-server-side-using-node-js)) to generate PNGs of arbitrary size from SVG source files server side. This would resolve the security concern, and provides a lot of leverage for solving the issues of aesthetics and responsiveness.

**Comment by bcoe** [comment]
@seldo @kevinSuttle having a service that hosts SVGs, but in turn serves PNG of arbitrary sizes would be my preference:
- it lets us handle various screen sizes and resolutions.
- it starts to solve Laurie's security concerns.
- it keeps the schema so darn simple, just a string URL for the logo :+1:

**Comment by mathiasbynens** [comment]
@cure53 Is there a secure way to display user-supplied SVGs?

My thinking:
1. npmjs.org should include `` rather than inlining the SVG or using data URLs, because `` never executes scripts embedded in `foo.svg`.
2. Of course, browsing to `foo.svg` directly would execute the scripts, so the SVG files should be hosted on another, dedicated origin (GitHub Pages-style).

→ ``

**Comment by bcoe** [comment]
I've moved the conversation here: https://github.com/npm/npm-explicit-installs/issues/13

**Comment by marcellodesales** [comment]
Awesome!

**Pull request merged**